### ISSUE TITLE
Plugin handler needs reorganized/rewrite

### ISSUE THREAD

**Issue by raluvy95** [raluvy95/jolly#7]
Currently all plugins are handled in hardcoded imports which isn't that bad at that time when Jolly was quite small. Now Jolly has grown and the hardcoded handler makes the code look stupid.

I attempted to create `JollyPlugin` class which has similar purpose to command handler, except that it relies on events + other type such as failure on command handler. Sadly, I abandoned it due to unable to understanding type manipulation or know how to get function arguments from EventHandler interface.

So in conclusion, we stuck on using hardcoded plugin imports, which you can see right now at `src/classes/events.ts`.
If you know how type manipultion works and know how to fix it. I would really appreciate from you :)
